exporters/committime/collector_base.py

```diff
@@ -236,16 +238,17 @@ def __parse_repourl(self):
         """Parse the repo_url into individual pieces"""
         if self.__repo_url is None:
             return
-
-        url_tokens = self.__repo_url.split("/")
-        self.__repo_protocol = url_tokens[0]
-        if self.__repo_protocol.endswith(':'):
-            self.__repo_protocol = self.__repo_protocol[:-1]
-        # token 1 is always a blank
-        self.__repo_fqdn = url_tokens[2]
-        self.__repo_group = url_tokens[3]
-        self.__repo_name = url_tokens[4]
-        if self.__repo_name.endswith('.git'):
-            self.__repo_project = self.__repo_name[:-4]
+        parsed = giturlparse.parse(self.__repo_url)
+        if len(parsed.protocols) > 0 and parsed.protocols[0] not in CommitMetric.supported_protocols:
+            raise Exception("Unsupported protocol %s", parsed.protocols[0])
```

Comment:
Can we raise a ValueError here, instead of a generic exception, and handle that specifically in the catch?

Reply:
sure

Reply:
ValueError is in the latest revision